caikit_nlp/modules/text_generation/text_generation.py

```diff
@@ -0,0 +1,261 @@
+)
+from .text_generation_task import TextGenerationTask
+
+log = alog.use_channel("TXT_GEN")
+error = error_handler.get(log)
+
+
+# pylint: disable=too-many-lines,too-many-instance-attributes
+@module(
+    id="f9181353-4ccf-4572-bd1e-f12bcda26792",
+    name="Text Generation",
+    version="0.1.0",
+    backend_type=TGISBackend.backend_type,
+    task=TextGenerationTask,
+)
+class TextGeneration(ModuleBase):
+    """Module to provide text generation capabilities"""
+
+    SUPPORTED_LOAD_BACKENDS = [TGISBackend.backend_type, backend_types.LOCAL]
+
+    supported_resources = [HFAutoCausalLM, HFAutoSeq2SeqLM]
+
+    def __init__(
+        self,
+        base_model_name: str,
+        base_model: PretrainedModelBase = None,
+        bos_token: str = None,
+        sep_token: str = None,
+        eos_token: str = None,
+        pad_token: str = None,
+        tgis_backend: TGISBackend = None,
+    ):
+        super().__init__()
+
+        error.type_check("<FPT00609194E>", str, allow_none=True, bos_token=bos_token)
+        error.type_check("<FPT72469403E>", str, allow_none=True, sep_token=sep_token)
+        error.type_check("<FPT48137045E>", str, allow_none=True, eos_token=eos_token)
+        error.type_check("<FPT53511308E>", str, allow_none=True, pad_token=pad_token)
+        self.base_model = base_model
+        self.base_model_name = base_model_name
+
+        # Set _model_loaded as False by default. This will only get set to True if
+        # we enable the tgis_backend and we are able to fetch the client successfully.
+        self._model_loaded = False
+        # Configure the internal client
+        # NOTE: This is made optional for the cases where we do not need to execute `.run` function
+        # for example, bootstrapping a model to caikit format and saving.
+        if tgis_backend:
+            self._client = tgis_backend.get_client(base_model_name)
+            # mark that the model is loaded so that we can unload it later
+            self._model_loaded = True
+
+        self._bos_token = bos_token
+        self._sep_token = sep_token
+        self._eos_token = eos_token
+        self._pad_token = pad_token
+
+    def __del__(self):
+        # nothing to unload if we didn't finish loading
+        if self._model_loaded:
+            self.get_backend().unload_model(self._model_path)
+
+    @classmethod
+    def bootstrap(cls, base_model_path: str, load_backend: BackendBase = None):
+        """Function to bootstrap a pre-trained transformers model and
+        get a caikit text-generation 'model'.
+
+        Args:
+            base_model_path: str
+                Path to transformers model
+                NOTE: Model path needs to contain tokenizer as well
+            load_backend: BackendBase
+                Backend object to be used to run inference with.
+                NOTE: this is required for inferencing. It is
+                made optional just in provide support for model conversion use-case
```

Comment:
```suggestion
                made optional to support the model conversion use-case
```
By model conversion, you mean to ORT etc, right? Could you elaborate a bit more on what this looks like?

I had was under the impression that TGIS model conversions generally happen in memory, and there's not an export API, so I don't think I understand how you'd leverage that without inference support

Reply:
The model conversion here refers to both converting the model if required by TGIS (like ORT) or just converting the model into `caikit` format, since at load time this module (in runtime) will only support loading of a `caikit` model and not download / load models that are in `huggingface` format